str.go

```diff
@@ -0,0 +1,78 @@
+package merkletree
+
+import (
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	treeRoot    *interiorNode
+	epoch       int64
+	prevEpoch   int64
+	prevStrHash []byte
+	sig         []byte
+	policies    []byte
+	prev        *SignedTreeRoot
+}
+
+func (m *MerkleTree) generateSTR(ep int64, prevEp int64, prevHash []byte) *SignedTreeRoot {
+	bytesPreSig := getSTRBytesForSig(m, ep, prevEp, prevHash)
+	sig := crypto.Sign(m.privKey, bytesPreSig)
+
+	return &SignedTreeRoot{
+		treeRoot:    m.root,
+		epoch:       ep,
+		prevEpoch:   prevEp,
+		prevStrHash: prevHash,
+		sig:         sig,
+		policies:    m.policies.Serialize(),
+		prev:        nil,
+	}
+}
+
+func (m *MerkleTree) generateNextSTR(ep int64) *SignedTreeRoot {
+	currentSTR := getCurrentSTR()
+	prevEpoch := currentSTR.epoch
+	prevStrHash := crypto.Digest(serializeSTR(*currentSTR))
+	bytesPreSig := getSTRBytesForSig(m, ep, prevEpoch, prevStrHash)
```

Comment:
From this point on, it looks like the same thing as `generateSTR`. Can it use that?